src/Microsoft.CodeAnalysis.BannedApiAnalyzers/Core/SymbolIsBannedAnalyzer.cs

```diff
@@ -181,7 +179,8 @@ private void OnCompilationStart(CompilationStartAnalysisContext compilationConte
             {
                 var query =
                     from additionalFile in compilationContext.Options.AdditionalFiles
-                    where StringComparer.Ordinal.Equals(Path.GetFileName(additionalFile.Path), SymbolIsBannedAnalyzer.BannedSymbolsFileName)
+                    let fileName = Path.GetFileName(additionalFile.Path)
+                    where fileName != null && fileName.StartsWith("BannedSymbols.", StringComparison.Ordinal) && fileName.EndsWith(".txt", StringComparison.Ordinal)
```

Comment:
Do we want to allow `BannedSymbols..txt`? (which is currently allowed with your implementation).

Reply:
I'd say yes, since it matches the `BannedSymbols.*.txt` pattern. I don't see a good reason to special-case it.